fmriprep/interfaces/fmap.py

```diff
@@ -322,11 +322,20 @@ def get_ees(in_meta, in_file=None):
     if ees is not None:
         return ees
 
-    # All other cases require the parallel acc and npe (N vox in PE dir)
     acc = float(in_meta.get('ParallelReductionFactorInPlane', 1.0))
     npe = nb.load(in_file).shape[_get_pe_index(in_meta)]
     etl = npe // acc
 
+    # Prefer metadata over size of the nifti file (Philips data can have
+    # different number of acquired versus reconstructed k-space lines
+    # see https://github.com/poldracklab/fmriprep/issues/1029 )
+
+    if 'EPIFactor' in in_meta:
+        etl = in_meta['EPIFactor'] + 1  # ETL is EPIFactor + 1 for Philips
+
+    # Not part of the BIDS spec now
```

Comment:
is this key under revision? I can remember suggesting it at some point, but I don't know where it stands at this point in time.

Reply:
And another question: does `EchoTrainLength` make sense if no `EPIFactor` is defined? (i.e., should next line be under this if branch?)